src/editors/sharedComponents/TinyMceWidget/hooks.js

```diff
@@ -151,6 +156,28 @@ export const setupCustomBehavior = ({
       editor, images, setImage, openImgModal,
     }),
   });
+
+  // insert link button
+  /* addButton only has setDisabled/isDisabled in api while
+    addToggleButton has setDisabled/isDisabled/setActive/isActive
+    addButton doc: https://www.tiny.cloud/docs/ui-components/typesoftoolbarbuttons/#basicbutton
+    addToggleButton: doc: https://www.tiny.cloud/docs/ui-components/typesoftoolbarbuttons/#togglebutton
+  */
+  editor.ui.registry.addToggleButton(tinyMCE.buttons.insertLink, {
+    icon: 'new-tab',
+    tooltip: translations?.insertLinkTooltipTitle ?? '',
+    onAction: openInsertLinkModal,
+    onSetup(api) {
+      editor.on('SelectionChange', () => {
+        const node = editor.selection.getNode();
+        const isLink = node.nodeName === 'A';
+        const hasTextSelected = editor.selection.getContent().length > 0;
+        api.setActive(isLink);
+        api.setDisabled(!isLink && !hasTextSelected);
+      });
```

Comment:
```suggestion
      editor.on('SelectionChange', () => {
        const node = editor.selection.getNode();
        
        const isLink = node.nodeName === 'A';
        let isInternalLink = false;
        if (isLink) {
          const selectedHTML = editor.selection.getContent({ format: 'html' }) || node.outerHTML;
          const regexDataBlockId = new RegExp(/data-block-id="([^"]+)"/);
          isInternalLink = regexDataBlockId.test(selectedHTML);
        }

        const hasTextSelected = editor.selection.getContent().length > 0;
        api.setActive(isInternalLink);
        api.setDisabled(!isInternalLink && !hasTextSelected);
      });
```

By doing this I was able to get it so the "jump to" button is only enabled when trying to edit internal links, but I was able to break it decently quickly (and uncovered another bug in the process).

The native link editor doesn't remove `data-block-id`, so by
* Creating an internal link using the "jump to" dialog
* Editing that internal link and changing it to be an external link using the native link dialog

We end up with a link on the page with an associated block id even though the url doesn't match anymore.

This gets cleaned up when opening the "jump to" menu during the block id/url validation, but I'd like to clean it up before getting in there.

I'm looking into ways to do that without duplicating too much code.